eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/FailureMessage.scala

```diff
@@ -44,7 +44,9 @@ sealed trait FailureMessage {
 sealed trait BadOnion extends FailureMessage { def onionHash: ByteVector32 }
 sealed trait Perm extends FailureMessage
 sealed trait Node extends FailureMessage
-sealed trait Update extends FailureMessage { def update: ChannelUpdate }
+// Historically, this trait guaranteed that a channel update was provided.
+// This was changed in https://github.com/lightning/bolts/pull/1163.
+sealed trait Update extends FailureMessage { def update: Option[ChannelUpdate] }
```

Comment:
Nit, for consistency with our established convention:
```suggestion
sealed trait Update extends FailureMessage { def update_opt: Option[ChannelUpdate] }
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2854/commits/2816203aa0e32f709e298cc7f013a7ee2d556f03